pkg/minishift/openshift/registry.go

```diff
@@ -25,6 +25,44 @@ import (
 )
 
 func GetDockerRegistryInfo() (string, error) {
+	var registryInfo string
+	var err error
+	if isRegistryRouteEnable() {
+		registryInfo, err = fetchRegistryRoute()
+	} else {
+		registryInfo, err = fetchRegistryService()
+	}
+	return registryInfo, err
+}
+
+func isRegistryRouteEnable() bool {
```

Comment:
> There are wrong assumptions about cause and effect. You use oc to see whether there is a route docker-registry and then you basically conclude that this route is created by the add-on and hence later on you append the '443' port. What if the user exposed the route in a different way by manual configuration?

@hferentschik , yes in that case this is problematic but what if a user use `addon apply` does that recorded by `AddOnManager` I didn't test it if that is true then will modify.